wgpu-hal/src/dx12/adapter.rs

```diff
@@ -181,6 +181,37 @@ impl super::Adapter {
             hr == 0 && features3.CastingFullyTypedFormatSupported != 0
         };
 
+        let shader_model = {
+            let mut sm: crate::dx12::types::D3D12_FEATURE_DATA_SHADER_MODEL =
+                unsafe { mem::zeroed() };
```

Comment:
> before calling the function initialize the HighestShaderModel field to the highest shader model that your application understands. After the function completes successfully, the HighestShaderModel field contains the highest shader model that is both supported by the device and no higher than the shader model passed in.

I don't think this is correct given what the D3D12 docs say.

This is further complicated by:

 > ID3D12Device::CheckFeatureSupport returns E_INVALIDARG if HighestShaderModel isn't known by the current runtime. For that reason, we recommend that you call this in a loop with decreasing shader models to determine the highest supported shader model.

There is an example here: https://github.com/microsoft/DirectXTK12/wiki/Shader-Model-6